seonshine_backend/models/signUpModel.js

```diff
@@ -0,0 +1,104 @@
+import { DataTypes, Model } from "sequelize";
+import { sequelizeUserDb } from "../db/dbConfig.js";
+
+class SignUp extends Model {}
+
+SignUp.init(
+  {
+    user_id: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notNull: {
+          msg: "User ID is required",
+        },
+        len: {
+          args: [8, 255],
+          msg: "User ID must be between 8 and 255 characters long",
+        },
+      },
+    },
+    role_id: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notNull: {
+          msg: "Role ID is required",
+        },
+      },
+    },
+    username: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notNull: {
+          msg: "Username is required",
+        },
+      },
+    },
+    phone_number: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notNull: {
+          msg: "Phone number is required",
+        },
+        isNumeric: {
+          msg: "Phone number must contain only numbers",
+        },
+      },
+    },
+    email: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notNull: {
+          msg: "Email is required",
+        },
+        isEmail: {
+          msg: "Invalid email address",
+        },
+      },
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notNull: {
+          msg: "Password is required",
+        },
+        len: {
+          args: [6, 255],
+          msg: "Password must be between 6 and 255 characters long",
+        },
+      },
+    },
+    branch_id: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      validate: {
+        isRequiredForRoleId1(value) {
```

Comment:
Should use meaning name instead of number. You can use `isRequiredWithNormalUser`. In this case, just use `isRequiredBasedRole`